sample/src/main/kotlin/com/slack/circuit/sample/repo/PetRepository.kt

```diff
@@ -19,41 +19,28 @@ import com.slack.circuit.sample.data.Animal
 import com.slack.circuit.sample.data.PetfinderApi
 import javax.inject.Inject
 import javax.inject.Singleton
-import kotlinx.coroutines.DelicateCoroutinesApi
-import kotlinx.coroutines.GlobalScope
-import kotlinx.coroutines.flow.MutableStateFlow
-import kotlinx.coroutines.flow.StateFlow
-import kotlinx.coroutines.launch
 
 interface PetRepository {
-  val animalsStateFlow: StateFlow<List<Animal>>
-  fun getAnimal(id: Long): Animal
+  suspend fun getAnimals(): List<Animal>
+  suspend fun getAnimal(id: Long): Animal?
```

Comment:
I reworked PetRepository to get rid of the animal StateFlow. I was having trouble getting the updated state to propagate through to the presenter as needed. Given that the presenter itself already maintained its own state object, and the repo was just using a coroutine under the hood, I reworked both methods to be suspending functions. This simplified things and got me to the expected behaviour both in tests and in app.